archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/gradle/FieldSelectorFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2014-2022 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.junit.gradle;
+
+import com.tngtech.archunit.junit.FieldSource;
+import com.tngtech.archunit.junit.ReflectionUtils;
+import org.gradle.api.internal.tasks.testing.junitplatform.TestSelectorFactory;
+import org.junit.platform.engine.TestSource;
+
+public class FieldSelectorFactory implements TestSelectorFactory {
+    @Override
+    public boolean supports(TestSource testSource) {
+        return testSource.getClass().getName().equals(FieldSource.class.getName());
```

Comment:
I needed to use reflection here due to some classloader issues

(don't even ask what tricks I needed to play on Gradle to even load this class...)